app/controllers/concerns/polymorphic_resource_scope.rb

```diff
@@ -81,10 +81,70 @@ def check_polymorphic_controller_resources
   end
 
   def _resource_ids
-    params["#{ polymorphic_klass_name }_id"] || super
+    return params["#{ polymorphic_klass_name }_id"] if params["#{ polymorphic_klass_name }_id"]
+
+    if respond_to?(:resource_name) && params.has_key?("#{ resource_name }_id")
+      params["#{ resource_name }_id"]
+    elsif params.has_key?(:id)
+      params[:id]
+    else
+      ''
+    end
   end
 
   def no_resources_error_message
     "Could not find #{polymorphic_klass_name} #{controller_name} with #{no_resources_message_ids}"
   end
+
+  def check_controller_resources
```

Comment:
I've basically copied over all the code from `RoleControl::Owned` to this file, and merged any method overrides (inlined `Owned` methods where this file called `super`). 

I did this because it was really difficult to see how the two interplayed, since it keeps `controlled_resources` but adds `polymorphic_*` methods alongside it. Except that some of the `controlled_*` methods *are* overridden, and it's a mess to figure out how those fit together (e.g. which methods from `Owned` are actually still needed? Do we actually call both versions sometimes?). Didn't want to think about this while also thinking about the overall refactor.

To keep PR size down, I'd like to leave it like this for now, and refactor the polymorphic stuff to policy objects in a future PR.

Reply:
Makes sense - ideally it is just a different policy object resolving to the polymorphic scopes.